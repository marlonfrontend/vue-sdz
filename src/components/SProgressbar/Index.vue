<template>
  <div class="s-progressbar">
    <div class="progress" :style="percent" />
  </div>
</template>

<script>
export default {
  name: 'SProgressbar',

  props: {
    percent: Number
  }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-progressbar {
  display: flex;
  background-color: #ECEDEF;
  height: 8px;
  width: 100%;
  border-radius: 8px;

  & > .progress{
    background-color: color(positive, base);
    border-radius: 8px;
  }
}
</style>
