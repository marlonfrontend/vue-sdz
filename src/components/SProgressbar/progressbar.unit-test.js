import { mount } from '@vue/test-utils'

import SProgressbar from './Index.vue'

/* eslint-disable */
describe('SProgressbar', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(SProgressbar, {
      propsData:{
        value: 65
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check class', () => {

    const cl = cmp.find('div')

    expect(cl.classes()).toContain('s-progress-bar')
  })

  test('check class', () => {

    const cl = cmp.find('div > div')

    expect(cl.attributes().style).toContain('width')
  })

})
