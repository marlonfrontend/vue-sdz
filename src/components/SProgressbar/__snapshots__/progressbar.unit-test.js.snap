// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SProgressbar has the expected html structure 1`] = `
<div
  class="s-progress-bar"
>
  <div
    class="progress"
    style="width: 65%;"
  />
</div>
`;
